abi/src/test/java/org/web3j/abi/DefaultFunctionEncoderTest.java

```diff
@@ -70,6 +59,144 @@ public void testBuildMethodSignatureWithStructWithArray() {
+
+        List<TypeReference<?>> MyStruct2Types = new ArrayList<>();
+        List<TypeReference<?>> MyStructTypes = new ArrayList<>();
+        List<TypeReference<?>> MyParameters = new ArrayList<>();
+
+        MyStruct2Types.add(TypeReference.makeTypeReference("string"));
+        MyStruct2Types.add(TypeReference.makeTypeReference("string"));
+
+        MyStructTypes.add(TypeReference.makeTypeReference("uint256"));
+        MyStructTypes.add(TypeReference.makeTypeReference("string"));
+        MyStructTypes.add(TypeReference.makeTypeReference("string"));
+        MyStructTypes.add(new TypeReference<DynamicStruct>(false, MyStruct2Types) {});
+
+        MyParameters.add(TypeReference.makeTypeReference("string"));
+        MyParameters.add(new TypeReference<DynamicStruct>(false, MyStructTypes) {});
+
+        MyParameters.add(TypeReference.makeTypeReference("string"));
+
+        List<Type> decodedData =
+                FunctionReturnDecoder.decode(returnedData, Utils.convert(MyParameters));
+
+        assertEquals(decodedData.get(0).getValue(), "valuebefore");
+
+        List<Type> structData = ((DynamicStruct) decodedData.get(1)).getValue();
+
+        assertEquals(structData.get(0).getValue(), BigInteger.valueOf(1));
+        assertEquals(structData.get(1).getValue(), "2");
+        assertEquals(structData.get(2).getValue(), "3");
+
+        List<Type> innerStructData = ((DynamicStruct) structData.get(3)).getValue();
+
+        assertEquals(innerStructData.get(0).getValue(), "1234");
+        assertEquals(innerStructData.get(1).getValue(), "0x1234");
+
+        assertEquals(decodedData.get(2).getValue(), "valueafter");
+    }
+
+    @Test
+    public void testArrayOfDynamicStruct() throws ClassNotFoundException {
+        // The full event signature is
+        //
+        //     Stamp3(uint256 indexed stampId, address indexed caller, bool odd,
+        // (uint256,bool,string) topMessage, (uint256,bool,string)[] messages),
+        //
+        // but we are only decoding the non-indexed data portion of it represented by
+        // 'bool odd, (uint256,bool,string) topMessage, (uint256,bool,string)[] messages'.
+        //
+        // Transaction:
+        // https://testnet.treasurescan.io/tx/0x041e53e7571283d462df99a95b2c21324279657f26a3adef907095d2d9c5ed85?tab=logs
+        // Contract:
+        // https://testnet.treasurescan.io/address/0x5167E9A422aCEd95C2D0b62bF05a7847a9a942B2
```

Comment:
please create a simpler contract example  an deploy it on sepolia as above example, this one is tricky because it shows first the interface and developers will need to search a bit more to find the struct

Reply:
I changed the test method name, it can be keep as it is